tooling/miniutil/src/build/statement.rs

```diff
@@ -1,5 +1,29 @@
 use crate::build::*;
 
+impl FunctionBuilder {
+    pub fn assign(&mut self, destination: PlaceExpr, source: ValueExpr) {
+        self.statements.push(Statement::Assign { destination, source });
+    }
+
+    pub fn set_discriminant(&mut self, destination: PlaceExpr, value: impl Into<Int>) {
+        self.statements.push(Statement::SetDiscriminant { destination, value: value.into() });
+    }
+
+    pub fn validate(&mut self, place: PlaceExpr, fn_entry: bool) {
+        self.statements.push(Statement::Validate { place, fn_entry });
+    }
+
+    pub fn storage_live(&mut self, local: PlaceExpr) {
+        let PlaceExpr::Local(name) = local else { panic!("PlaceExpr is not a local") };
+        self.statements.push(Statement::StorageLive(name));
+    }
+
+    pub fn storage_dead(&mut self, local: PlaceExpr) {
```

Comment:
```suggestion
    #[track_caller]
    pub fn storage_live(&mut self, local: PlaceExpr) {
        let PlaceExpr::Local(name) = local else { panic!("PlaceExpr is not a local") };
        self.statements.push(Statement::StorageLive(name));
    }

    #[track_caller]
    pub fn storage_dead(&mut self, local: PlaceExpr) {
```
That will make the panics a bit more informative